core/common/src/main/java/alluxio/concurrent/ManagedBlockingUfsForwarder.java

```diff
@@ -0,0 +1,654 @@
+        return mUfs.renameDirectory(src, dst);
+      }
+    }.get();
+  }
+
+  @Override
+  public boolean renameRenamableDirectory(String src, String dst) throws IOException {
+    return new ManagedBlockingUfsMethod<Boolean>() {
+      @Override
+      public Boolean execute() throws IOException {
+        return mUfs.renameRenamableDirectory(src, dst);
+      }
+    }.get();
+  }
+
+  @Override
+  public boolean renameFile(String src, String dst) throws IOException {
+    return new ManagedBlockingUfsMethod<Boolean>() {
+      @Override
+      public Boolean execute() throws IOException {
+        return mUfs.renameFile(src, dst);
+      }
+    }.get();
+  }
+
+  @Override
+  public boolean renameRenamableFile(String src, String dst) throws IOException {
+    return new ManagedBlockingUfsMethod<Boolean>() {
+      @Override
+      public Boolean execute() throws IOException {
+        return mUfs.renameRenamableFile(src, dst);
+      }
+    }.get();
+  }
+
+  @Override
+  public AlluxioURI resolveUri(AlluxioURI ufsBaseUri, String alluxioPath) {
+    return mUfs.resolveUri(ufsBaseUri, alluxioPath);
+  }
+
+  @Override
+  public void setAclEntries(String path, List<AclEntry> aclEntries) throws IOException {
+    new ManagedBlockingUfsFunction() {
+      @Override
+      public void execute() throws IOException {
+        mUfs.setAclEntries(path, aclEntries);
+      }
+    }.get();
+  }
+
+  @Override
+  public void setMode(String path, short mode) throws IOException {
+    new ManagedBlockingUfsFunction() {
+      @Override
+      public void execute() throws IOException {
+        mUfs.setMode(path, mode);
+      }
+    }.get();
+  }
+
+  @Override
+  public void setOwner(String path, String owner, String group) throws IOException {
+    new ManagedBlockingUfsFunction() {
+      @Override
+      public void execute() throws IOException {
+        mUfs.setOwner(path, owner, group);
+      }
+    }.get();
+  }
+
+  @Override
+  public boolean supportsFlush() {
+    return mUfs.supportsFlush();
+  }
+
+  @Override
+  public boolean supportsActiveSync() {
+    return mUfs.supportsActiveSync();
+  }
+
+  @Override
+  public SyncInfo getActiveSyncInfo() throws IOException {
+    return mUfs.getActiveSyncInfo();
+  }
+
+  @Override
+  public void startSync(AlluxioURI uri) throws IOException {
+    mUfs.startSync(uri);
+  }
+
+  @Override
+  public void stopSync(AlluxioURI uri) throws IOException {
+    mUfs.stopSync(uri);
+  }
+
+  @Override
+  public boolean startActiveSyncPolling(long txId) throws IOException {
+    return mUfs.startActiveSyncPolling(txId);
+  }
+
+  @Override
+  public boolean stopActiveSyncPolling() throws IOException {
+    return mUfs.stopActiveSyncPolling();
+  }
+
+  @Override
+  public void close() throws IOException {
+    mUfs.close();
+  }
+
+  /**
+   * Utility class used to isolate calls into underlying UFS from concurrency compensation logic.
+   * Note: This class used to make calls with a return value.
```

Comment:
Could `Void` be used for the calls without a return value?

Reply:
We would need to return "null" (or anything) from void methods, like below:

```
@Override
  public void setAclEntries(String path, List<AclEntry> aclEntries) throws IOException {
    new ManagedBlockingUfsMethod<Void>() {
      @Override
      public Void execute() throws IOException {
        mUfs.setAclEntries(path, aclEntries);
        return null;
      }
    }.get();
}
```

Reply:
Right, I prefer this over having two separate classes, WDYT?

Reply:
Agreed.